src/factories/nodeBar.ts

```diff
@@ -0,0 +1,49 @@
+import { differenceInMilliseconds, millisecondsInSecond } from 'date-fns'
+import { Container } from 'pixi.js'
+import { DEFAULT_LINEAR_COLUMN_SIZE_PIXELS, DEFAULT_TIME_COLUMN_SIZE_PIXELS } from '@/consts'
+import { barFactory } from '@/factories/bar'
+import { RunGraphNode } from '@/models/RunGraph'
+import { waitForConfig } from '@/objects/config'
+import { layout } from '@/objects/layout'
+
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export async function nodeBarFactory() {
+  const config = await waitForConfig()
+  const { bar, render: renderBar } = await barFactory()
+
+  async function render(node: RunGraphNode): Promise<Container> {
+    const { background = '#fff' } = config.styles.node(node)
+    const { nodeHeight: height, nodeBorderRadius: radius } = config.styles
+    const width = getTotalWidth(node, radius)
+    const capLeft = node.state_type !== 'running'
+
+    await renderBar({
+      width,
+      height,
+      radius,
+      background,
+      capLeft,
+    })
+
+    return bar
+  }
+
+  function getTotalWidth(node: RunGraphNode, borderRadius: number): number {
+    if (layout.horizontal === 'trace') {
+      const right = node.start_time
+      const left = node.end_time ?? new Date()
+      const seconds = differenceInMilliseconds(left, right) / millisecondsInSecond
+      const width = seconds * DEFAULT_TIME_COLUMN_SIZE_PIXELS
+
+      // this means the min node size is 18px. Is that correct?
+      return Math.max(width, borderRadius * 2)
```

Comment:
I believe `borderRadius` is set to 8px, so the min width would be 16px. Glad you caught the min-width there, key for some level of visibility / clickability.

Reply:
Right it's 2x the border radius. Not 18px. That tracts with what you'd expect?

Reply:
Yeah totally